src/utilities/utilities.cpp

```diff
@@ -463,6 +463,9 @@ std::string GetDeviceArchitecture(const Device& device) {
     else if (device.HasExtension(kKhronosAttributesAMD)) {
       device_architecture = device.Name(); // Name is architecture for AMD APP and AMD ROCm
     }
+    else if (device.IsQualcomm()) { // queries the Adreno GPU architecture version
```

Comment:
Similar to my other comment above, should we add a `device.isGPU()` to make sure this only applies to GPU Qualcomm devices? The `AdrenoVersion` function's name hints at the fact that this is GPU-only.